### PR TITLE
User who edited the document can't approve the document

### PR DESCRIPTION
**Pull request by AndrewVos**
https://trello.com/c/R9gwpJap/297-disallow-the-author-of-the-change-to-approve-the-changes

### CODE REVIEW

File: spec/models/edition_spec.rb [hunk 1]
```diff
@@ -89,16 +89,28 @@
...
       it "returns true when a review has been requested" do
         edition.state = "review_requested"
         edition.save!
-        expect(edition.can_be_approved?).to be true
+        expect(edition.can_be_approved?(user)).to be true
+      end
+
+      it "returns false when the user is also the editor" do
+        edition.state = "review_requested"
+        edition.user = User.create!(name: "anotehr")
```
Comment (tadast) [inline spec/models/edition_spec.rb]: typo

File: app/models/edition.rb [hunk 2]
```diff
@@ -48,8 +48,8 @@ def can_request_review?
     true
   end
 
-  def can_be_approved?
-    persisted? && review_requested?
+  def can_be_approved? by_user
```
Comment (tadast) [inline app/models/edition.rb]: Missing parenthesis in method definition #styleguide

File: spec/models/edition_spec.rb (hunk 1 above)
Comment (AndrewVos) [inline spec/models/edition_spec.rb]: no it's not

**Comment by tadast** [comment]
LGTM

**Pull request merged**